driver/src/test/java/org/neo4j/driver/internal/InternalStatementResultTest.java

```diff
@@ -386,23 +386,89 @@ public void shouldNotPeekIntoTheFutureWhenResultIsEmpty()
         Record future = result.peek();
     }
 
+    @Test
+    public void shouldNotifyResourcesHandlerWhenFetchedViaList()
+    {
+        SessionResourcesHandler resourcesHandler = mock( SessionResourcesHandler.class );
+        StatementResult result = createResult( 10, resourcesHandler );
+
+        List<Record> records = result.list();
+        assertEquals( 10, records.size() );
+
+        verify( resourcesHandler ).onResultConsumed();
+    }
+
+    @Test
+    public void shouldNotifyResourcesHandlerWhenFetchedViaSingle()
+    {
+        SessionResourcesHandler resourcesHandler = mock( SessionResourcesHandler.class );
+        StatementResult result = createResult( 1, resourcesHandler );
+
+        Record record = result.single();
+        assertEquals( "v1-1", record.get( "k1" ).asString() );
+
+        verify( resourcesHandler ).onResultConsumed();
+    }
+
+    @Test
+    public void shouldNotifyResourcesHandlerWhenFetchedViaIterator()
+    {
+        SessionResourcesHandler resourcesHandler = mock( SessionResourcesHandler.class );
+        StatementResult result = createResult( 1, resourcesHandler );
+
+        while ( result.hasNext() )
+        {
+            assertNotNull( result.next() );
+        }
+
+        verify( resourcesHandler ).onResultConsumed();
+    }
+
+    @Test
+    public void shouldNotifyResourcesHandlerWhenSummary()
+    {
+        SessionResourcesHandler resourcesHandler = mock( SessionResourcesHandler.class );
+        StatementResult result = createResult( 10, resourcesHandler );
+
+        assertNotNull( result.summary() );
```

Comment:
Added a separate test for this here: https://github.com/neo4j/neo4j-java-driver/pull/324/commits/0ddac3d5036b7017f80f6592ba4815b0b00bcd87